src/pres/Settings.js

```diff
@@ -1,11 +1,62 @@
 import React from 'react';
 import { Button, Col, Container, Row } from 'reactstrap';
 import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
-import { faDesktop, faFastBackward, faMoon, faLightbulb, faRetweet, faTrashAlt } from '@fortawesome/free-solid-svg-icons';
+import { faDesktop, faFastBackward, faMoon, faLightbulb, faRetweet, faTrashAlt, faBookOpen } from '@fortawesome/free-solid-svg-icons';
+import axios from 'axios';
+import useSWR from 'swr';
 
 import { trackEvent } from '../app/Analytics';
 import * as Constants from '../app/Constants';
 
+function OpeningTheory({openingManager}) {
+    const lastMoveNumber = Math.ceil(openingManager.currentIndex / 2)
+
+    const {data: url} = useSWR(openingManager.currentIndex > 0 ? `Chess Opening Theory/${openingManager.pgnListSoFar().slice(0, lastMoveNumber).map(({whitePly, blackPly, moveNumber}) => {
```

Comment:
This code could be made more readable. Its not clear what the API request request/response is. Which part of it we are using etc.